signer/src/testing/mod.rs

```diff
@@ -112,3 +116,13 @@ where
 }
 
 impl<I, T> IterTestExt<T> for I where I: IntoIterator<Item = T> + Sized {}
+
+/// Returns a seedable rng with random seed. Prints the seed to
+/// stderr so that it can be used to reproduce the test
+pub fn get_rng() -> StdRng {
+    let seed = OsRng.next_u64();
+
+    // Nextest prints stdout and stderr only for failing tests
+    eprintln!("Failed with seed: {seed}");
```

Comment:
I might change this to the following, just because `eprintln`'s do show up if you run the test in other ways (e.g. with `--nocapture` etc.).

```suggestion
    // Nextest prints stderr only for failing tests, so this message
    // will only appear if the test fails (by default).
    eprintln!("Test executed with seed: {seed}");
```